pennylane/gradients/gradient_transform.py

```diff
@@ -0,0 +1,178 @@
+
+
+def gradient_expand(tape, depth=10):
+    """Expand out a tape so that it supports differentiation
+    of requested operations.
+
+    This is achieved by decomposing all trainable operations that have
+    ``Operation.grad_method=None`` until all resulting operations
+    have a defined gradient method.
+
+    Args:
+        tape (.QuantumTape): the input tape to expand
+        depth (int) : the maximum expansion depth
+
+    Returns:
+        .QuantumTape: the expanded tape
+    """
+
+    # check if the tape contains unsupported trainable operations
+    if any(unsupported_op(op) and trainable_op(op) for op in tape.operations):
+
+        # Define the stopping condition for the expansion
+        stop_cond = lambda obj: (
+            not isinstance(obj, qml.measure.MeasurementProcess)
+            and ((supported_op(obj) and trainable_op(obj)) or not trainable_op(obj))
+        )
+
+        return tape.expand(depth=depth, stop_at=stop_cond)
+
+    return tape
+
+
+class gradient_transform(qml.batch_transform):
+    """Decorator for defining quantum gradient transforms.
+
+    Quantum gradient transforms are a specific case of :class:`~.batch_transform`.
+    All quantum gradient transforms accept a tape, and outputs
+    a batch of tapes to be independently executed on a quantum device, alongside
+    a post-processing function that returns the result.
+
+    Args:
+        expand_fn (function): An expansion function (if required) to be applied to the
+            input tape before the gradient computation takes place. If not provided,
+            the default expansion function simply expands all operations that
+            have ``Operation.grad_method=None`` until all resulting operations
+            have a defined gradient method.
+        differentiable (bool): Specifies whether the gradient transform is differentiable or
+            not. A transform may be non-differentiable if it does not use an
+            autodiff framework for its tensor manipulations. In such a case, setting
+            ``differentiable=False`` instructs the decorator
+            to mark the output as 'constant', reducing potential overhead.
+        hybrid (bool): Specifies whether classical processing inside a QNode
+            should be taken into account when transforming a QNode.
+
+            - If ``True``, and classical processing is detected and this
+              option is set to ``True``, the Jacobian of the classical
+              processing will be computed and included. When evaluated, the
+              returned Jacobian will be with respect to the QNode arguments.
+
+            - If ``False``, any internal QNode classical processing will be
+              **ignored**. When evaluated, the returned Jacobian will be with
+              respect to the **gate** arguments, and not the QNode arguments.
+
+    Supported gradient transforms must be of the following form:
+
+    .. code-block:: python
+
+        @quantum_gradient

```

Comment:
```suggestion
        @gradient_transform
```
?

Reply:
thanks, this was a previous name I was trying out before I decided I didn't like it 😆